### ISSUE TITLE
New internal test app fails on regeneration

### ISSUE THREAD

**Issue by summer-cook** [samvera/hyrax#6125]
### Descriptive summary

A new `internal_test_app` (created by `engine_cart` gem) is not generated upon adding migrations - instead, a cached version of the internal test app is used unless a change to the `.circleci/config.yml` or `regen` file is added. This is intentionally how it is set up, but prevents the test suite from knowing about new migrations since it is using a cached version of the test app even when a change to the schema is added. Due to this, we were running into an issue that even when a test passes locally, it fails in CI due to an error that the PG Table does not exist.

Subsequently, upon making a change to the regen file to force the test app to generate a new version, the build fails.

It seems to be failing on running the following command within `engine_cart` gem
`(bin/rails g | grep test_app) && bin/rails g test_app`
aka, it is failing to generate the test_app.

We were able to replicate this error on a clean branch with only an update to the `.regen` file.
https://github.com/samvera/hyrax/pull/6129

### Steps to reproduce the behavior in User Interface (UI)

1. Not testable in UI - this is an error happening in CI
2. Use the CI from the replicated issue here: https://github.com/samvera/hyrax/pull/6129
3. SSH into the Circle CI build
4. cd into the `.internal_test_app`
5. run the commands in the engine cart rake task at beginning here: https://github.com/cbeer/engine_cart/blob/ef1270f2b0ef7bea0d5b22e81975881f40e5b5f4/lib/engine_cart/tasks/engine_cart.rake#L112 this is the area where it is failing.

### Actual behavior (include screenshots if available)
Example failure when regen file is not updated/test app is being cached:
```
 1) Hyrax::CounterMetric validations is valid with valid attributes
     Failure/Error: counter_metric = build(:counter_metric)
     
     ActiveRecord::StatementInvalid:
       PG::UndefinedTable: ERROR:  relation "hyrax_counter_metrics" does not exist
       LINE 8:  WHERE a.attrelid = '"hyrax_counter_metrics"'::regclass
                                   ^
```

Example error when not cached/generating a new internal test app. Happens in the main branch CI

<img width="633" alt="image" src="https://github.com/samvera/hyrax/assets/73361970/d5c622a4-4257-48e1-8061-0509cea4d8e7">

### Acceptance Criteria/Expected Behavior

- [ ] If a new table is added to the DB, the internal test app detects the change, and the new database structure is reflected in the test suite. 
- [ ] A new internal test app can be generated successfully

### Related work

This is the original PR where the failure was occurring
- https://github.com/samvera/hyrax/pull/6124

Clean PR replication to work off of:
- https://github.com/samvera/hyrax/pull/6129

**Comment by dlpierce** [comment]
I think the cause of this is the addition of a `bin/rails` script in #6123. I removed it and touched .regen and the internal_test_app rebuilt. CI run results are [here](https://app.circleci.com/pipelines/github/samvera/hyrax/9349). It appears this script is also causing problems running Nurax PG in a local docker compose stack, so we need to find an alternative. 
@orangewolf ^^

**Comment by orangewolf** [comment]
closed by #6146
